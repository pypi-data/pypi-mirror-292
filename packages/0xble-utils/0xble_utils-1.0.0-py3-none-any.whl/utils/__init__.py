from .strings import *
from .cli import *
from .time import *
