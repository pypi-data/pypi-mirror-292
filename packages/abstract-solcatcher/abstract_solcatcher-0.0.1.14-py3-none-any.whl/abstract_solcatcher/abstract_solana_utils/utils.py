from solders.pubkey import Pubkey
import base64

def get_pubString(obj):
    return Pubkey.from_string(str(obj))

def get_pubBytes(obj):
    return Pubkey.from_bytes(obj)

def get_addres_lookup_table(txnData):
    return txnData.get('transaction', {}).get('message', {}).get('addressTableLookups', [])

def get_account_keys(txnData):
    return txnData.get('transaction', {}).get('message', {}).get('accountKeys', [])

def get_loaded_addresses(txnData):
    return txnData['meta']['loadedAddresses']

def get_read_only_addresses(txnData):
    return get_loaded_addresses(txnData).get('readonly', [])

def get_writable_addresses(txnData):
    return get_loaded_addresses(txnData).get('writable', [])

def get_log_messages(txnData):
    return txnData['meta']['logMessages']

def get_instructions(txnData):
  return txnData['transaction']['message']['instructions']

def get_inner_instructions(txnData):
  return txnData['meta']['innerInstructions'][0]['instructions']

def update_instructions(txnData,instructions):
  txnData['transaction']['message']['instructions'] = instructions
  return  txnData

def update_inner_instructions(txnData,inner_instructions):
  txnData['meta']['innerInstructions'][0]['instructions'] = inner_instructions
  return txnData

def get_post_balance_fromm_txn(txnData):
    return txnData.get('meta', {}).get('postTokenBalances', [])
def get_pre_balance_fromm_txn(txnData):
    return txnData.get('meta', {}).get('preTokenBalances', [])
def get_all_account_keys(txnData):
  accountKeys=[]
  accountKeys += get_account_keys(txnData)
  accountKeys += get_read_only_addresses(txnData)
  accountKeys += get_writable_addresses(txnData)
  return accountKeys

def search_for_account_index(data,index_number):
    for index_data in data:
        if str(index_data.get('accountIndex')) == str(index_number):
            return index_data
