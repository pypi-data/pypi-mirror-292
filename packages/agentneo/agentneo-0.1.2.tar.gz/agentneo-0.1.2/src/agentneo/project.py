from .agent_neo import AgentNeo
from typing import List, Dict, Any


class Project:
    def __init__(self, session: AgentNeo, project_name: str, description: str):
        self.session = session
        self.project_name = project_name
        self.description = description
        self.project_id = None

    def create(self) -> Dict[str, Any]:
        try:
            response = self.session._make_request(
                "POST",
                "projects/create_project",
                data={"name": self.project_name, "description": self.description},
            )
            if "id" in response:
                self.project_id = response["id"]
                print(
                    f"Project '{self.project_name}' created successfully "
                    f"with ID: {self.project_id}"
                )
            else:
                print(
                    f"Failed to create project '{self.project_name}'. "
                    f"Response: {response}"
                )
            return response
        except Exception as e:
            print(f"An error occurred while creating the project: {str(e)}")
            return {"error": str(e)}

    @classmethod
    def list_projects(cls, session: AgentNeo) -> List[Dict[str, Any]]:
        projects = session._make_request("GET", "projects/get_projects")
        if not projects:
            print("No projects available.")
            return
        print("Available projects:")
        for project in projects:
            print(f"- {project['name']}")
        return

    @staticmethod
    def list_metrics(session: AgentNeo) -> List[Dict[str, Any]]:
        metrics = session._make_request("GET", "metrics/get_metrics")
        if not metrics:
            print("No metrics available.")
            return
        print("Available metrics:")
        for metric in metrics:
            print(f"- {metric['name']}")
        return
