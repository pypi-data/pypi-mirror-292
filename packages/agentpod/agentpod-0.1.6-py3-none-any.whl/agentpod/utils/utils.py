def normalize_to_id(input_string: str) -> str:
    """
    Converts a normal sentence into a normalized ID format by replacing spaces with underscores,
    removing special characters, converting to lowercase, and appending a timestamp to ensure uniqueness.

    Args:
        input_string (str): The input string to be normalized.

    Returns:
        str: A normalized string suitable for use as an ID.
    """
    import re
    import time

    # Replace spaces with underscores
    input_string = input_string.replace(" ", "_")
    # Remove all non-alphanumeric characters (except underscores) and convert to lowercase
    normalized_string = re.sub(r"[^\w_]+", "", input_string).lower()
    # Append current timestamp
    timestamp = int(time.time())
    unique_normalized_string = f"{normalized_string}_{timestamp}"
    return unique_normalized_string


def escape_curly_braces(input_string: str) -> str:
    """
    Escapes curly braces in the input string so it can be safely used in an f-string formatter.

    Args:
        input_string (str): The input string that may contain curly braces.

    Returns:
        str: The input string with curly braces escaped.
    """
    return input_string.replace("{", "{{").replace("}", "}}")
