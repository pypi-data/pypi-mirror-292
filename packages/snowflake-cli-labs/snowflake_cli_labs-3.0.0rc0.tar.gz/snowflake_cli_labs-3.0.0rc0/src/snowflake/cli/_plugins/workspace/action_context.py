from dataclasses import dataclass
from pathlib import Path


@dataclass
class ActionContext:
    """
    An object that is passed to each action when called by WorkspaceManager
    """

    project_root: Path
