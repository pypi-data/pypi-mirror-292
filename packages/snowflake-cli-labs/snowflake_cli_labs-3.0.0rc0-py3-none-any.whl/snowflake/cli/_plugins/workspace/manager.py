from pathlib import Path
from typing import Dict

from snowflake.cli._plugins.workspace.action_context import ActionContext
from snowflake.cli.api.entities.common import EntityActions
from snowflake.cli.api.exceptions import InvalidProjectDefinitionVersionError
from snowflake.cli.api.project.schemas.entities.entities import (
    Entity,
    v2_entity_model_to_entity_map,
)
from snowflake.cli.api.project.schemas.project_definition import (
    DefinitionV20,
    ProjectDefinition,
)


class WorkspaceManager:
    """
    Instantiates entity instances from entity models, providing higher-order functionality on entity compositions.
    """

    def __init__(self, project_definition: ProjectDefinition, project_root: Path):
        if not project_definition.meets_version_requirement("2"):
            raise InvalidProjectDefinitionVersionError(
                "2.x", project_definition.definition_version
            )
        self._entities_cache: Dict[str, Entity] = {}
        self._project_definition: DefinitionV20 = project_definition
        self._project_root = project_root

    def get_entity(self, entity_id: str):
        """
        Returns an entity instance with the given ID. If exists, reuses the previously returned instance, or instantiates a new one otherwise.
        """
        if entity_id in self._entities_cache:
            return self._entities_cache[entity_id]
        entity_model = self._project_definition.entities.get(entity_id, None)
        if entity_model is None:
            raise ValueError(f"No such entity ID: {entity_id}")
        entity_model_cls = entity_model.__class__
        entity_cls = v2_entity_model_to_entity_map[entity_model_cls]
        self._entities_cache[entity_id] = entity_cls(entity_model)
        return self._entities_cache[entity_id]

    def perform_action(self, entity_id: str, action: EntityActions):
        """
        Instantiates an entity of the given ID and calls the given action on it.
        """
        entity = self.get_entity(entity_id)
        if entity.supports(action):
            action_ctx = ActionContext(project_root=self.project_root())
            return entity.perform(action, action_ctx)
        else:
            raise ValueError(f'This entity type does not support "{action.value}"')

    def project_root(self) -> Path:
        return self._project_root
