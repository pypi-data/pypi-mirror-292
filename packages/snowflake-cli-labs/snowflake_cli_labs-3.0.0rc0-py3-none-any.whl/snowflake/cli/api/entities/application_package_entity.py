from contextlib import contextmanager
from pathlib import Path
from textwrap import dedent
from typing import List, Optional

from click import ClickException
from snowflake.cli._plugins.nativeapp.artifacts import build_bundle
from snowflake.cli._plugins.nativeapp.bundle_context import BundleContext
from snowflake.cli._plugins.nativeapp.codegen.compiler import NativeAppCompiler
from snowflake.cli._plugins.nativeapp.constants import (
    ALLOWED_SPECIAL_COMMENTS,
    COMMENT_COL,
    INTERNAL_DISTRIBUTION,
    NAME_COL,
    SPECIAL_COMMENT,
)
from snowflake.cli._plugins.nativeapp.exceptions import (
    ApplicationPackageAlreadyExistsError,
)
from snowflake.cli._plugins.workspace.action_context import ActionContext
from snowflake.cli.api.console.abc import AbstractConsole
from snowflake.cli.api.entities.common import EntityBase, get_sql_executor
from snowflake.cli.api.entities.utils import (
    ensure_correct_owner,
    generic_sql_error_handler,
    render_script_templates,
)
from snowflake.cli.api.exceptions import SnowflakeSQLExecutionError
from snowflake.cli.api.project.schemas.entities.application_package_entity_model import (
    ApplicationPackageEntityModel,
)
from snowflake.cli.api.rendering.jinja import (
    jinja_render_from_str,
)
from snowflake.connector import ProgrammingError


class ApplicationPackageEntity(EntityBase[ApplicationPackageEntityModel]):
    """
    A Native App application package.
    """

    def action_bundle(self, ctx: ActionContext):
        model = self._entity_model
        bundle_map = build_bundle(
            ctx.project_root, Path(model.deploy_root), model.artifacts
        )
        bundle_context = BundleContext(
            package_name=model.identifier,
            artifacts=model.artifacts,
            project_root=ctx.project_root,
            bundle_root=Path(model.bundle_root),
            deploy_root=Path(model.deploy_root),
            generated_root=Path(model.generated_root),
        )
        compiler = NativeAppCompiler(bundle_context)
        compiler.compile_artifacts()
        return bundle_map

    @staticmethod
    def get_existing_app_pkg_info(
        package_name: str,
        package_role: str,
    ) -> Optional[dict]:
        """
        Check for an existing application package by the same name as in project definition, in account.
        It executes a 'show application packages like' query and returns the result as single row, if one exists.
        """
        sql_executor = get_sql_executor()
        with sql_executor.use_role(package_role):
            return sql_executor.show_specific_object(
                "application packages", package_name, name_col=NAME_COL
            )

    @staticmethod
    def get_app_pkg_distribution_in_snowflake(
        package_name: str,
        package_role: str,
    ) -> str:
        """
        Returns the 'distribution' attribute of a 'describe application package' SQL query, in lowercase.
        """
        sql_executor = get_sql_executor()
        with sql_executor.use_role(package_role):
            try:
                desc_cursor = sql_executor.execute_query(
                    f"describe application package {package_name}"
                )
            except ProgrammingError as err:
                generic_sql_error_handler(err)

            if desc_cursor.rowcount is None or desc_cursor.rowcount == 0:
                raise SnowflakeSQLExecutionError()
            else:
                for row in desc_cursor:
                    if row[0].lower() == "distribution":
                        return row[1].lower()
        raise ProgrammingError(
            msg=dedent(
                f"""\
                Could not find the 'distribution' attribute for application package {package_name} in the output of SQL query:
                'describe application package {package_name}'
                """
            )
        )

    @classmethod
    def verify_project_distribution(
        cls,
        console: AbstractConsole,
        package_name: str,
        package_role: str,
        package_distribution: str,
        expected_distribution: Optional[str] = None,
    ) -> bool:
        """
        Returns true if the 'distribution' attribute of an existing application package in snowflake
        is the same as the the attribute specified in project definition file.
        """
        actual_distribution = (
            expected_distribution
            if expected_distribution
            else cls.get_app_pkg_distribution_in_snowflake(
                package_name=package_name,
                package_role=package_role,
            )
        )
        project_def_distribution = package_distribution.lower()
        if actual_distribution != project_def_distribution:
            console.warning(
                dedent(
                    f"""\
                    Application package {package_name} in your Snowflake account has distribution property {actual_distribution},
                    which does not match the value specified in project definition file: {project_def_distribution}.
                    """
                )
            )
            return False
        return True

    @staticmethod
    @contextmanager
    def use_package_warehouse(
        package_warehouse: Optional[str],
    ):
        if package_warehouse:
            with get_sql_executor().use_warehouse(package_warehouse):
                yield
        else:
            raise ClickException(
                dedent(
                    f"""\
                Application package warehouse cannot be empty.
                Please provide a value for it in your connection information or your project definition file.
                """
                )
            )

    @classmethod
    def apply_package_scripts(
        cls,
        console: AbstractConsole,
        package_scripts: List[str],
        package_warehouse: Optional[str],
        project_root: Path,
        package_role: str,
        package_name: str,
    ) -> None:
        """
        Assuming the application package exists and we are using the correct role,
        applies all package scripts in-order to the application package.
        """

        if package_scripts:
            console.warning(
                "WARNING: native_app.package.scripts is deprecated. Please migrate to using native_app.package.post_deploy."
            )

        queued_queries = render_script_templates(
            project_root,
            jinja_render_from_str,
            dict(package_name=package_name),
            package_scripts,
        )

        # once we're sure all the templates expanded correctly, execute all of them
        with cls.use_package_warehouse(
            package_warehouse=package_warehouse,
        ):
            try:
                for i, queries in enumerate(queued_queries):
                    console.step(f"Applying package script: {package_scripts[i]}")
                    get_sql_executor().execute_queries(queries)
            except ProgrammingError as err:
                generic_sql_error_handler(
                    err, role=package_role, warehouse=package_warehouse
                )

    @classmethod
    def create_app_package(
        cls,
        console: AbstractConsole,
        package_name: str,
        package_role: str,
        package_distribution: str,
    ) -> None:
        """
        Creates the application package with our up-to-date stage if none exists.
        """

        # 1. Check for existing existing application package
        show_obj_row = cls.get_existing_app_pkg_info(
            package_name=package_name,
            package_role=package_role,
        )

        if show_obj_row:
            # 1. Check for the right owner role
            ensure_correct_owner(
                row=show_obj_row, role=package_role, obj_name=package_name
            )

            # 2. Check distribution of the existing application package
            actual_distribution = cls.get_app_pkg_distribution_in_snowflake(
                package_name=package_name,
                package_role=package_role,
            )
            if not cls.verify_project_distribution(
                console=console,
                package_name=package_name,
                package_role=package_role,
                package_distribution=package_distribution,
                expected_distribution=actual_distribution,
            ):
                console.warning(
                    f"Continuing to execute `snow app run` on application package {package_name} with distribution '{actual_distribution}'."
                )

            # 3. If actual_distribution is external, skip comment check
            if actual_distribution == INTERNAL_DISTRIBUTION:
                row_comment = show_obj_row[COMMENT_COL]

                if row_comment not in ALLOWED_SPECIAL_COMMENTS:
                    raise ApplicationPackageAlreadyExistsError(package_name)

            return

        # If no application package pre-exists, create an application package, with the specified distribution in the project definition file.
        sql_executor = get_sql_executor()
        with sql_executor.use_role(package_role):
            console.step(f"Creating new application package {package_name} in account.")
            sql_executor.execute_query(
                dedent(
                    f"""\
                    create application package {package_name}
                        comment = {SPECIAL_COMMENT}
                        distribution = {package_distribution}
                """
                )
            )
