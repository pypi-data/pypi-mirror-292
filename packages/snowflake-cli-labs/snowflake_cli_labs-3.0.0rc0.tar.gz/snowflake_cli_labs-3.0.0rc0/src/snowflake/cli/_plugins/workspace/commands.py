# Copyright (c) 2024 Snowflake Inc.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
# http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

from __future__ import annotations

import logging

import typer
import yaml
from click import ClickException
from snowflake.cli._plugins.nativeapp.artifacts import BundleMap
from snowflake.cli._plugins.snowpark.commands import migrate_v1_snowpark_to_v2
from snowflake.cli._plugins.streamlit.commands import migrate_v1_streamlit_to_v2
from snowflake.cli._plugins.workspace.manager import WorkspaceManager
from snowflake.cli.api.cli_global_context import get_cli_context
from snowflake.cli.api.commands.decorators import with_project_definition
from snowflake.cli.api.commands.snow_typer import SnowTyper
from snowflake.cli.api.entities.common import EntityActions
from snowflake.cli.api.output.types import MessageResult
from snowflake.cli.api.project.definition_manager import DefinitionManager
from snowflake.cli.api.secure_path import SecurePath

ws = SnowTyper(
    name="ws",
    help="Deploy and interact with snowflake.yml-based entities.",
)
log = logging.getLogger(__name__)


@ws.command()
def migrate(
    accept_templates: bool = typer.Option(
        False, "-t", "--accept-templates", help="Allows the migration of templates."
    ),
    **options,
):
    """Migrates the Snowpark and Streamlit project definition files form V1 to V2."""
    pd = DefinitionManager().unrendered_project_definition

    if pd.meets_version_requirement("2"):
        return MessageResult("Project definition is already at version 2.")

    if "<% ctx." in str(pd):
        if not accept_templates:
            raise ClickException(
                "Project definition contains templates. They may not be migrated correctly, and require manual migration."
                "You can try again with --accept-templates  option, to attempt automatic migration."
            )
        log.warning(
            "Your V1 definition contains templates. We cannot guarantee the correctness of the migration."
        )

    if pd.streamlit:
        pd_v2 = migrate_v1_streamlit_to_v2(pd)
    elif pd.snowpark:
        pd_v2 = migrate_v1_snowpark_to_v2(pd)
    else:
        raise ValueError(
            "Only Snowpark and Streamlit entities are supported for migration."
        )

    SecurePath("snowflake.yml").rename("snowflake_V1.yml")
    with open("snowflake.yml", "w") as file:
        yaml.dump(
            pd_v2.model_dump(
                exclude_unset=True, exclude_none=True, mode="json", by_alias=True
            ),
            file,
        )
    return MessageResult("Project definition migrated to version 2.")


@ws.command(requires_connection=True, hidden=True)
@with_project_definition()
def validate(
    **options,
):
    """Validates the project definition file."""
    # If we get to this point, @with_project_definition() has already validated the PDF schema
    return MessageResult("Project definition is valid.")


@ws.command(requires_connection=True, hidden=True)
@with_project_definition()
def bundle(
    entity_id: str = typer.Option(
        help=f"""The ID of the entity you want to bundle.""",
    ),
    **options,
):
    """
    Prepares a local folder with the configured artifacts of the specified entity.
    """

    cli_context = get_cli_context()
    ws = WorkspaceManager(
        project_definition=cli_context.project_definition,
        project_root=cli_context.project_root,
    )

    bundle_map: BundleMap = ws.perform_action(entity_id, EntityActions.BUNDLE)
    return MessageResult(f"Bundle generated at {bundle_map.deploy_root()}")
