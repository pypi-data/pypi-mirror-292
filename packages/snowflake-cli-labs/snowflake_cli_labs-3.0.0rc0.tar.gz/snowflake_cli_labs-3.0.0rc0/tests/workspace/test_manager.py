# Copyright (c) 2024 Snowflake Inc.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
# http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
from __future__ import annotations

import logging
import os
from pathlib import Path
from unittest import mock

import pytest
from snowflake.cli._plugins.workspace.manager import WorkspaceManager
from snowflake.cli.api.entities.common import EntityActions
from snowflake.cli.api.exceptions import InvalidProjectDefinitionVersionError
from snowflake.cli.api.project.definition_manager import DefinitionManager

from tests.testing_utils.files_and_dirs import create_named_file
from tests.workspace.utils import (
    APP_PACKAGE_ENTITY,
    MOCK_SNOWFLAKE_YML_FILE,
    MOCK_SNOWFLAKE_YML_V1_FILE,
)


def _get_ws_manager(pdf_content=MOCK_SNOWFLAKE_YML_FILE):
    current_working_directory = os.getcwd()
    create_named_file(
        file_name="snowflake.yml",
        dir_name=current_working_directory,
        contents=[pdf_content],
    )
    dm = DefinitionManager()
    return WorkspaceManager(
        project_definition=dm.project_definition,
        project_root=dm.project_root,
    )


def test_pdf_not_v2(temp_dir):
    with pytest.raises(InvalidProjectDefinitionVersionError):
        _get_ws_manager(pdf_content=MOCK_SNOWFLAKE_YML_V1_FILE)


# Test that the same entity instance is returned for the same id
def test_get_entity_is_cached(temp_dir):
    ws_manager = _get_ws_manager()
    pkg1 = ws_manager.get_entity("pkg")
    pkg2 = ws_manager.get_entity("pkg")
    app = ws_manager.get_entity("app")
    assert pkg1 is pkg2
    assert app is not pkg1


def test_get_entity_invalid_id(temp_dir):
    ws_manager = _get_ws_manager()
    with pytest.raises(ValueError, match="No such entity ID"):
        ws_manager.get_entity("non_existing_id")


def test_bundle(temp_dir):
    ws_manager = _get_ws_manager()
    with mock.patch(f"{APP_PACKAGE_ENTITY}.action_bundle") as app_pkg_bundle_mock:
        ws_manager.perform_action("pkg", EntityActions.BUNDLE)
        app_pkg_bundle_mock.assert_called_once()


def test_bundle_of_invalid_entity_type(temp_dir):
    ws_manager = _get_ws_manager()
    with pytest.raises(
        ValueError, match='This entity type does not support "action_bundle"'
    ):
        ws_manager.perform_action("app", EntityActions.BUNDLE)


@pytest.mark.parametrize(
    "project_directory_name",
    ["migration_streamlit_V1_to_V2", "migration_snowpark_V1_to_V2"],
)
def test_migration_v1_to_v2(
    runner, project_directory, snapshot, project_directory_name
):
    with project_directory(project_directory_name):
        result = runner.invoke(["ws", "migrate"])

    assert result.exit_code == 0
    assert "Project definition migrated to version 2." in result.output
    assert Path("snowflake.yml").read_text() == snapshot
    assert Path("snowflake_V1.yml").read_text() == snapshot


@pytest.mark.parametrize(
    "project_directory_name", ["migration_streamlit_V2", "migration_snowpark_V2"]
)
def test_migration_already_v2(runner, project_directory, project_directory_name):
    with project_directory(project_directory_name):
        result = runner.invoke(["ws", "migrate"])

    assert result.exit_code == 0
    assert "Project definition is already at version 2." in result.output


@pytest.mark.parametrize(
    "project_directory_name", ["snowpark_templated_v1", "streamlit_templated_v1"]
)
def test_if_template_is_not_rendered_during_migration_with_option_checked(
    runner, project_directory, project_directory_name, os_agnostic_snapshot, caplog
):
    with project_directory(project_directory_name):
        with caplog.at_level(logging.WARNING):
            result = runner.invoke(["ws", "migrate", "--accept-templates"])

    assert result.exit_code == 0
    assert Path("snowflake.yml").read_text() == os_agnostic_snapshot
    assert Path("snowflake_V1.yml").read_text() == os_agnostic_snapshot
    assert (
        "Your V1 definition contains templates. We cannot guarantee the correctness of the migration."
        in caplog.text
    )


@pytest.mark.parametrize(
    "project_directory_name", ["snowpark_templated_v1", "streamlit_templated_v1"]
)
def test_if_template_raises_error_during_migrations(
    runner, project_directory, project_directory_name, os_agnostic_snapshot
):
    with project_directory(project_directory_name):
        result = runner.invoke(["ws", "migrate"])
        assert result.exit_code == 1
        assert "Project definition contains templates" in result.output
