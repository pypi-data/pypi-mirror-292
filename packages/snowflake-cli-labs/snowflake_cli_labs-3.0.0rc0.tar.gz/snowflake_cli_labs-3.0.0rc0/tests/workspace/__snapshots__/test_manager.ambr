# serializer version: 1
# name: test_if_template_is_not_rendered_during_migration_with_option_checked[snowpark_templated_v1]
  '''
  definition_version: '2'
  entities:
    hello_function:
      artifacts:
      - dest: <! project_name | to_snowflake_identifier !>
        src: <% ctx.env.project_source %>
      external_access_integrations: []
      handler: functions.hello_function
      identifier:
        name: hello_function
      imports: []
      meta:
        use_mixins:
        - snowpark_shared
      returns: string
      secrets: {}
      signature:
      - name: name
        type: string
      stage: <! stage | to_snowflake_identifier !>
      type: function
    hello_procedure:
      artifacts:
      - dest: <! project_name | to_snowflake_identifier !>
        src: <% ctx.env.project_source %>
      execute_as_caller: false
      external_access_integrations: []
      handler: procedures.hello_procedure
      identifier:
        name: hello_procedure
      imports: []
      meta:
        use_mixins:
        - snowpark_shared
      returns: string
      secrets: {}
      signature:
      - name: name
        type: string
      stage: <! stage | to_snowflake_identifier !>
      type: procedure
    test_procedure:
      artifacts:
      - dest: <! project_name | to_snowflake_identifier !>
        src: <% ctx.env.project_source %>
      execute_as_caller: false
      external_access_integrations: []
      handler: procedures.test_procedure
      identifier:
        name: test_procedure
      imports: []
      meta:
        use_mixins:
        - snowpark_shared
      returns: string
      secrets: {}
      signature: ''
      stage: <! stage | to_snowflake_identifier !>
      type: procedure
  env:
    project_source: app/
  mixins:
    snowpark_shared:
      artifacts:
      - dest: <! project_name | to_snowflake_identifier !>
        src: <% ctx.env.project_source %>
      stage: <! stage | to_snowflake_identifier !>
  
  '''
# ---
# name: test_if_template_is_not_rendered_during_migration_with_option_checked[snowpark_templated_v1].1
  '''
  definition_version: "1.1"
  snowpark:
    project_name: "<! project_name | to_snowflake_identifier !>"
    stage_name: "<! stage | to_snowflake_identifier !>"
    src: <% ctx.env.project_source %>
    functions:
      - name: hello_function
        handler: "functions.hello_function"
        signature:
          - name: "name"
            type: "string"
        returns: string
    procedures:
      - name: hello_procedure
        handler: "procedures.hello_procedure"
        signature:
          - name: "name"
            type: "string"
        returns: string
      - name: test_procedure
        handler: "procedures.test_procedure"
        signature: ""
        returns: string
  env:
    project_source: "app/"
  
  '''
# ---
# name: test_if_template_is_not_rendered_during_migration_with_option_checked[streamlit_templated_v1]
  '''
  definition_version: '2'
  entities:
    <! name | to_snowflake_identifier !>:
      artifacts:
      - streamlit_app.py
      - environment.yml
      - pages
      - common/hello.py
      identifier:
        name: <! name | to_snowflake_identifier !>
      main_file: streamlit_app.py
      pages_dir: pages
      query_warehouse: <! query_warehouse | to_snowflake_identifier !>
      stage: <! stage | to_snowflake_identifier !>
      title: <% ctx.env.streamlit_title %>
      type: streamlit
  env:
    streamlit_title: My Fancy Streamlit
  
  '''
# ---
# name: test_if_template_is_not_rendered_during_migration_with_option_checked[streamlit_templated_v1].1
  '''
  definition_version: "1.1"
  streamlit:
    name: <! name | to_snowflake_identifier !>
    stage: <! stage | to_snowflake_identifier !>
    query_warehouse: <! query_warehouse | to_snowflake_identifier !>
    main_file: streamlit_app.py
    env_file: environment.yml
    pages_dir: pages/
    title: <% ctx.env.streamlit_title %>
    additional_source_files:
      - common/hello.py
  env:
    streamlit_title: "My Fancy Streamlit"
  
  '''
# ---
# name: test_migration_v1_to_v2[migration_snowpark_V1_to_V2]
  '''
  definition_version: '2'
  entities:
    func1:
      artifacts:
      - dest: my_snowpark_project
        src: app
      external_access_integrations: []
      handler: app.func1_handler
      identifier:
        name: func1
      imports: []
      meta:
        use_mixins:
        - snowpark_shared
      returns: string
      runtime: '3.10'
      secrets: {}
      signature:
      - default: default value
        name: a
        type: string
      - name: b
        type: variant
      stage: dev_deployment
      type: function
    procedureName:
      artifacts:
      - dest: my_snowpark_project
        src: app
      execute_as_caller: false
      external_access_integrations: []
      handler: hello
      identifier:
        name: procedureName
      imports: []
      meta:
        use_mixins:
        - snowpark_shared
      returns: string
      secrets: {}
      signature:
      - name: name
        type: string
      stage: dev_deployment
      type: procedure
  mixins:
    snowpark_shared:
      artifacts:
      - dest: my_snowpark_project
        src: app/
      stage: dev_deployment
  
  '''
# ---
# name: test_migration_v1_to_v2[migration_snowpark_V1_to_V2].1
  '''
  definition_version: 1
  snowpark:
    project_name: "my_snowpark_project"
    stage_name: "dev_deployment"
    src: "app/"
    functions:
      - name: func1
        handler: "app.func1_handler"
        signature:
          - name: "a"
            type: "string"
            default: "default value"
          - name: "b"
            type: "variant"
        returns: string
        runtime: 3.10
    procedures:
      - name: procedureName
        handler: "hello"
        signature:
          - name: "name"
            type: "string"
        returns: string
  
  '''
# ---
# name: test_migration_v1_to_v2[migration_streamlit_V1_to_V2]
  '''
  definition_version: '2'
  entities:
    test_streamlit:
      artifacts:
      - streamlit_app.py
      - environment.yml
      - pages
      identifier:
        name: test_streamlit
      main_file: streamlit_app.py
      pages_dir: None
      query_warehouse: test_warehouse
      stage: streamlit
      title: My Fancy Streamlit
      type: streamlit
  
  '''
# ---
# name: test_migration_v1_to_v2[migration_streamlit_V1_to_V2].1
  '''
  definition_version: 1
  streamlit:
    name: test_streamlit
    stage: streamlit
    query_warehouse: test_warehouse
    main_file: "streamlit_app.py"
    title: "My Fancy Streamlit"
  
  '''
# ---
